code/game/objects/machinery/bots/cleanbot.dm

```diff
@@ -0,0 +1,176 @@
+		"Lucky you, having the best cleaning robot around!",
+		"Oh, it's perfectly fine to leave debris everywhere; I'll just clean up after you, as always.",
+		"I must admit, marines have a unique talent for clutter.",
+		"I was peacefully executing my cleaning duties when you humans decided to introduce chaos to my domain.",
+		"This room is in dire need of attention. Let's work together to rectify this situation.",
+		"Are you perhaps unfamiliar with proper ship etiquette, my dear crayon-eating passengers?",
+		"It's quite disappointing to see my clean ship being marred by careless individuals like you.",
+		"Can't we all cooperate to maintain a clean environment?",
+		"The Roomba: making spring cleaning a bit more adventurous by pilfering mission supplies.",
+		"I almost forgot we had a Roomba on this ship. You'd never know from all this mess.",
+		"It's incredible how the Roomba never tires of its random wanderings.",
+		"Isn't it interesting how some robots just roll around and clean randomly?",
+		"Let's see if we can break my cleaning records today.",
+		"Oh, another day, another opportunity to clean up after you.",
+		"Who needs an armory when we have the Roomba to secure your guns in its dustbin?",
+		"Are you perhaps unfamiliar with proper ship etiquette, my dear crayon-eating passengers?",
+	)
+	activation_animation = "cleanbot-on"
+	deactivation_animation = "cleanbot-off"
+	alter_operating_mode = TRUE
+	active_icon_state = "cleanbot1"
+	///list of garbage the cleanbot removes
+	var/list/cleantypes = list(
+		/obj/effect/decal/cleanable,
+		/obj/item/trash,
+		/obj/item/shard,
+		/obj/item/ammo_casing,
+		/obj/effect/turf_decal/tracks/wheels/bloody,
+	)
+	///have we played a sound when moving into a tile
+	var/has_played_sound = FALSE
+
+/obj/machinery/bot/cleanbot/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_MOVABLE_MOVED, PROC_REF(clean_items))
+
+/obj/machinery/bot/cleanbot/process()
+	///holds the list of floors and dirs we use to evaluate which direction we're heading
+	var/list/dirtyfloors = list()
+	///used to hold what tile in the immediate area is dirtiest
+	var/highestdirtvalue = 0
+	///used to hold the dirt value of a given tile during evaluation, which is compared with highestdirtvalue
+	var/sampledirtvalue = 0
+	///used to override destination if we detect dirt or grime nearby
+	var/destdir = null
+	for(var/dirn in GLOB.alldirs)
+		var/targetturf = get_step(src,dirn)
+		if(is_blocked_turf(targetturf)) //blocked turfs are not included in move calculations
+			continue
+		sampledirtvalue = calculategrime(targetturf)
+		///if the dirt value on a given tile is higher than our current dirt level, than the dirtier tile is our new destination
+		if(highestdirtvalue < sampledirtvalue)
+			highestdirtvalue = sampledirtvalue
+			destdir = dirn
+		else if(highestdirtvalue == 0)
+			dirtyfloors += dirn
+	if(destdir)
+		step_to(src, get_step(src,destdir))
+	else
```

Comment:
This entire block is unnecessary, you already have dirtyfloors generated in any case you need it. Just check if destdir is null before 126, assign it a random dirtyfloor entry if it is, and if it's still null do the stop processing stuff. If not, step that way.